app/queries/changeset_query.py

```diff
@@ -124,3 +125,21 @@ async def find_many_by_query(
                 stmt = stmt.limit(limit)
 
             return (await session.scalars(stmt)).all()
+
+    @staticmethod
+    async def count_per_day_by_user_id(user_id: int, days: int) -> dict:
+        """
+        Count changesets per day by user id in given days past.
+        """
+        async with db() as session:
+            created_at_limit = utcnow() - timedelta(days=days)
+
+            stmt = (
+                select(
+                    func.date_trunc('day', Changeset.created_at).label('created_at_day_date'), func.count(Changeset.id)
+                )
+                .where(Changeset.user_id == user_id)
+                .where(Changeset.created_at > created_at_limit)
+                .group_by('created_at_day_date')
+            )
```

Comment:
You can move the date_trunc to a variable to make the query simpler:

```suggestion
            created_date = func.date_trunc('day', Changeset.created_at)
            stmt = (
                select(
                    created_date,
                    func.count(Changeset.id),
                )
                .where(Changeset.user_id == user_id)
                .where(created_date >= created_since)
                .group_by(created_date)
            )
```